src/pages/user/InstitutionDialog.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { useTranslation } from 'react-i18next';
+import InstitutionSelector from './InstitutionSelector';
+import styled from 'styled-components';
+import { useDispatch } from 'react-redux';
+import { User } from '../../types/user.types';
+import { updateInstitutionForAuthority } from './../../api/authorityApi';
+import { institutionLookup } from '../../api/institutionApi';
+import { setAuthorityData } from './../../redux/actions/userActions';
+import { addNotification } from '../../redux/actions/notificationActions';
+import { addInstitutionPresentation } from '../../redux/actions/institutionActions';
+
+const StyledInstitutionDialog = styled.div`
+  width: 20rem;
+`;
+
+interface InstitutionDialogProps {
+  user: User;
+  title: string;
+}
+
+const InstitutionDialog: React.FC<InstitutionDialogProps> = ({ user, title }) => {
+  const [open, setOpen] = useState(false);
+  const [selectedCristinUnitId, setSelectedCristinUnitId] = useState('');
+  const dispatch = useDispatch();
+  const { t } = useTranslation('profile');
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleCancel = () => {
+    setOpen(false);
+  };
+
+  const handleConfirm = async () => {
+    if (!user.authority.orgunitids?.find(orgunitid => orgunitid === selectedCristinUnitId)) {
+      const updatedAuthority = await updateInstitutionForAuthority(
+        selectedCristinUnitId,
+        user.authority.systemControlNumber
+      );
+      if (updatedAuthority?.error) {
+        dispatch(addNotification(updatedAuthority.error));
+      } else if (updatedAuthority) {
+        dispatch(setAuthorityData(updatedAuthority));
+        try {
+          const presentation = await institutionLookup(selectedCristinUnitId);
+          dispatch(addInstitutionPresentation(presentation));
+        } catch {}
+      }
+    }
+    setOpen(false);
+  };
+
+  return (
+    <StyledInstitutionDialog>
+      <Button
+        variant="contained"
+        color="primary"
+        onClick={handleClickOpen}
+        disabled={!user.authority?.systemControlNumber}>
+        {t('common:add')}
+      </Button>
+      <Dialog open={open} onClose={handleConfirm} aria-labelledby="form-dialog-title">
+        <DialogTitle id="form-dialog-title">{title}</DialogTitle>
+        <DialogContent>
+          <InstitutionSelector valueFunction={setSelectedCristinUnitId} />
```

Comment:
hva med å bare kalle propen for `setSelectedCristinUnitId` e.l.? `valueFunction` kan være hva som helst ut fra navnet